frontend/src/context/Context.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+} from "react";
+
+interface DataProviderProps<T extends object, F extends object> {
+  children: React.ReactNode;
+  fetchFunction: (filters: F) => Promise<T[]>;
+  initialFilters: F;
+}
+
+interface DataContextValue<T extends object, F extends object> {
+  data: T[];
+  fetchData: () => void;
+  filters: F;
+  setFilters: (filters: F) => void;
+}
+
+export const createDataContext = <T extends object, F extends object>() => {
+  const DataContext = createContext<DataContextValue<T, F> | undefined>(
+    undefined
+  );
+
+  const DataProvider: React.FC<DataProviderProps<T, F>> = ({
+    children,
+    fetchFunction,
+    initialFilters,
+  }) => {
+    const [data, setData] = useState<T[]>([]);
+    const [filters, setFilters] = useState<F>(initialFilters);
+
+    const fetchData = useCallback(async () => {
+      const result = await fetchFunction(filters);
```

Comment:
see my comment above for the logic behind this. and we can hardcode it for now but one of the tasks for this week is creating an AuthContext and wrapping the entire app with it, that way the user's company_id is accessible everywhere

```suggestion
      const result = await fetchFunction({...filters, company_id: "0a67f5d3-88b6-4e8f-aac0-5137b29917fd"});
```